x/concentrated-liquidity/internal/math/tick.go

```diff
@@ -1,52 +1,173 @@
+	// Due to sdk.Power restrictions, if the resulting power is negative, we take 9 * (1/10**exponentAtPriceOne)
+	geometricExponentIncrementDistanceInTicks := sdkNineDec.Mul(powTen(exponentAtPriceOne.Neg()))
+
+	// Initialize the total price to 1, the current precision to exponentAtPriceOne, and the number of ticks passed to 0
+	totalPrice := sdk.OneDec()
+	ticksPassed := sdk.ZeroInt()
+	exponentAtCurrentTick := exponentAtPriceOne
+
+	// Set the currentAdditiveIncrementInTicks to the exponentAtPriceOne
+	currentAdditiveIncrementInTicks := powTenBigDec(exponentAtPriceOne)
+
+	// Now, we loop through the k increments until we have passed the price
+	// Once we pass the price, we can determine what which geometric exponents we have filled in their entirety,
+	// as well as how many ticks that corresponds to
+	// In the opposite direction (price < 1), we do the same thing (just decrement the geometric exponent instead of incrementing).
+	// The only difference is we must reduce the increment distance by a factor of 10.
+	if price.GT(sdk.OneDec()) {
+		for totalPrice.LT(price) {
+			currentAdditiveIncrementInTicks = powTenBigDec(exponentAtCurrentTick)
+			maxPriceForcurrentAdditiveIncrementInTicks := osmomath.BigDecFromSDKDec(geometricExponentIncrementDistanceInTicks).Mul(currentAdditiveIncrementInTicks)
+			totalPrice = totalPrice.Add(maxPriceForcurrentAdditiveIncrementInTicks.SDKDec())
+			exponentAtCurrentTick = exponentAtCurrentTick.Add(sdk.OneInt())
+			ticksPassed = ticksPassed.Add(geometricExponentIncrementDistanceInTicks.TruncateInt())
+		}
+	} else {
+		// We must decrement the exponentAtCurrentTick by one when traversing negative ticks in order to constantly step up in precision when going further down in ticks
+		// Otherwise, from tick 0 to tick -(geometricExponentIncrementDistanceInTicks), we would use the same exponent as the exponentAtPriceOne
+		exponentAtCurrentTick := exponentAtPriceOne.Sub(sdk.OneInt())
+		for totalPrice.GT(price) {
+			currentAdditiveIncrementInTicks = powTenBigDec(exponentAtCurrentTick)
+			maxPriceForcurrentAdditiveIncrementInTicks := osmomath.BigDecFromSDKDec(geometricExponentIncrementDistanceInTicks).Mul(currentAdditiveIncrementInTicks)
+			totalPrice = totalPrice.Sub(maxPriceForcurrentAdditiveIncrementInTicks.SDKDec())
+			exponentAtCurrentTick = exponentAtCurrentTick.Sub(sdk.OneInt())
+			ticksPassed = ticksPassed.Sub(geometricExponentIncrementDistanceInTicks.TruncateInt())
+		}
+	}
+	// Determine how many ticks we have passed in the exponentAtCurrentTick
+	ticksToBeFulfilledByExponentAtCurrentTick := osmomath.BigDecFromSDKDec(price.Sub(totalPrice)).Quo(currentAdditiveIncrementInTicks)
+
+	// Finally, add the ticks we have passed from the completed geometricExponent values, as well as the ticks we have passed in the current geometricExponent value
+	tickIndex := ticksPassed.Add(ticksToBeFulfilledByExponentAtCurrentTick.SDKDec().TruncateInt())
```

Comment:
This section, notation, and comments were a bit unclear to me.

From my understanding, `ticksToBeFulfilledByExponentAtCurrentTick` is the `extraAdditiveTickDelta` that we went over by adding the full  `currentAdditiveIncrementInTicks` for that `expoenntAtCurrentTick` at the last iteration of the for loop.

I was also unable to interpret the comments to facilitate my understanding here. Please consider changing the identifier name and rephrasing comments.